src/managing-the-applications-lifecycle/manage-tech-debt/code-patterns/limit-records-sql.md

```diff
@@ -32,4 +32,11 @@ The solution is to control the number of rows inside the query:
 
 ![](images/limit-rows-sql.png)
 
-With this change you no longer rely on the  **&lt;Query&gt;.Count** property to get the correct count of rows complying with the original query conditions and joins.
+The SQL Version only works if you set the @maxrows input parameter to be an **Expand Inline** one. And that oblies you to change the data type of the query input parameter to text.
+**Expand Inline** parameters on your queries are not recommended, may be exploited and you may have findings on AI Mentor Studio:
+
+* [Learn more about AI Mentor Studio Findings](https://success.outsystems.com/documentation/11/managing_the_applications_lifecycle/manage_technical_debt/code_analysis_patterns/).
+
+In this particular case make sure to pass what you are passing to the query input @maxrows, if it's a variable or an input, through the **VerifySqlLiteral** function provided by the **Sanitization** extension.
+With this change you no longer rely on the **&lt;Query&gt;.Count** property to get the correct count of rows complying with the original query conditions and joins.
```

Comment:
Can you explain what this sentence means and how it relates to limiting the number of rows retrieved by a query?

Reply:
Yes. 
- So the SQL component always returns a count 
-  if you don't limit the query, the amount of records returned corresponds to the total amount, and to that count
- Imagine that you have a table with 100 records
- If you don't limit the query you get 100 records and a count of 100
- If you limit the query (for example with @maxrows) and you only fetch for example 10, you get 10 and the count will also be equals to 10
- Now, you need to tell the user that he is seeing 10 records out of 100
- To populate for example a List with pagination. 
-  And to get that number, 100, you need to do a separated count query to get how many records there are in the database that corresponds to the given criteria

Reply:
Super clear! Thanks a lot!